libs/auth/src/cryptography.ts

```diff
@@ -550,3 +550,44 @@ export async function signMessage({
     : [scriptPath, scriptInput];
   return runCommand(command, { stdin: message });
 }
+
+/**
+ * Generates a 256-bit random key using OpenSSL's rand command. The key is
+ * returned in hex encoding.
+ */
+export async function generateRandomAes256Key(): Promise<string> {
+  return (await openssl(['rand', String(32)])).toString('hex');
+}
+
+/**
+ * Encrypts the given utf8-encoded input string using AES-256-ECB with the
+ * specified hex-encoded key, returning the encrypted data in base64 encoding.
+ * To ensure this encryption is used securely, we want to output no more than a
+ * single block of data (128 bits of input).
+ */
+export async function encryptAes256(
+  key: string,
+  input: string
+): Promise<string> {
+  assert(Buffer.from(key, 'hex').length === 32, 'Key must be 256 bits');
+  const inputBuffer = Buffer.from(input, 'utf-8');
+  assert(inputBuffer.length <= 16, 'Input must be 128 bits or less');
+  return (
+    await openssl(['enc', '-aes-256-ecb', '-K', key, '-in', inputBuffer])
+  ).toString('base64');
+}
+
+/**
+ * Decrypts the given base64-encoded input string using AES-256-ECB with the
+ * specified hex-encoded key, returning the decrypted data as a utf8-encoded
+ * string.
+ */
+export async function decryptAes256(
+  key: string,
+  input: string
+): Promise<string> {
+  const inputBuffer = Buffer.from(input, 'base64');
```

Comment:
worth asserting that `inputBuffer` is just one block, so should be 128 bits exactly.

Reply:
Good call!